crates/valence_entity/src/active_status_effects.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+
+use bevy_ecs::prelude::*;
+
+use crate::status_effects::StatusEffect;
+
+/// [`Component`] that stores the [`ActiveStatusEffect`]s of an [`Entity`].
+#[derive(Component, Default, Debug)]
+pub struct ActiveStatusEffects {
+    active_effects: HashMap<StatusEffect, ActiveStatusEffect>,
+    new_effects: HashMap<StatusEffect, ActiveStatusEffect>,
+}
+
+impl ActiveStatusEffects {
+    pub fn add(&mut self, effect: ActiveStatusEffect) {
+        self.new_effects.insert(effect.status_effect(), effect);
+    }
+
+    pub fn active_effects(&self) -> &HashMap<StatusEffect, ActiveStatusEffect> {
+        &self.active_effects
+    }
+
+    pub fn active_effects_mut(&mut self) -> &mut HashMap<StatusEffect, ActiveStatusEffect> {
+        &mut self.active_effects
+    }
+
+    pub fn new_effects(&self) -> &HashMap<StatusEffect, ActiveStatusEffect> {
+        &self.new_effects
+    }
+
+    pub fn new_effects_mut(&mut self) -> &mut HashMap<StatusEffect, ActiveStatusEffect> {
+        &mut self.new_effects
+    }
+}
```

Comment:
This is what I would recommend using a Vec:
```suggestion
impl ActiveStatusEffects {
    pub fn add(&mut self, effect: ActiveStatusEffect) {
        if self.new_effects.iter().chain(self.active_effects.iter()).any(|(e, _)| e == &effect.status_effect()) {
            return;
        } // In a vector less then 100 elements, this is fast enough to justify not using a hashmap
    }

    pub fn iter(&self) -> impl Iterator<Item = &ActiveStatusEffect> {
        self.active_effects.iter().map(|(_, effect)| effect)
    }

    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut ActiveStatusEffect> {
        self.active_effects.iter_mut().map(|(_, effect)| effect)
    }

    pub fn new_effects(&self) -> impl Iterator<Item = &ActiveStatusEffect> {
        self.new_effects.iter().map(|(_, effect)| effect)
    }

    pub fn move_new_to_active(&mut self) -> impl Iterator<Item = &mut ActiveStatusEffect> {
        self.active_effects.extend( self.new_effects.drain(..));
        
        self.active_effects.iter_mut().map(|(_, effect)| effect)
    }
}
```

Reply:
oops made a mistake with the last one, use this:
```rs
pub fn move_new_to_active(&mut self) -> impl Iterator<Item = &mut ActiveStatusEffect> {
        let old_len = self.active_effects.len();
        self.active_effects.extend( self.new_effects.drain(..));

        self.active_effects[old_len..].iter_mut().map(|(_, effect)| effect)
    }
    ```